crates/primitives/src/chain_spec.rs

```diff
@@ -15,25 +17,30 @@ pub static MAINNET: Lazy<ChainSpec> = Lazy::new(|| ChainSpec {
         .expect("Can't deserialize Mainnet genesis json"),
     genesis_hash: H256(hex!("d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3")),
     hardforks: BTreeMap::from([
-        (Hardfork::Frontier, 0),
-        (Hardfork::Homestead, 1150000),
-        (Hardfork::Dao, 1920000),
-        (Hardfork::Tangerine, 2463000),
-        (Hardfork::SpuriousDragon, 2675000),
-        (Hardfork::Byzantium, 4370000),
-        (Hardfork::Constantinople, 7280000),
-        (Hardfork::Petersburg, 7280000),
-        (Hardfork::Istanbul, 9069000),
-        (Hardfork::Muirglacier, 9200000),
-        (Hardfork::Berlin, 12244000),
-        (Hardfork::London, 12965000),
-        (Hardfork::ArrowGlacier, 13773000),
-        (Hardfork::GrayGlacier, 15050000),
-        (Hardfork::Latest, 15050000),
+        (Hardfork::Frontier, ForkKind::Block(0)),
+        (Hardfork::Homestead, ForkKind::Block(1150000)),
+        (Hardfork::Dao, ForkKind::Block(1920000)),
+        (Hardfork::Tangerine, ForkKind::Block(2463000)),
+        (Hardfork::SpuriousDragon, ForkKind::Block(2675000)),
+        (Hardfork::Byzantium, ForkKind::Block(4370000)),
+        (Hardfork::Constantinople, ForkKind::Block(7280000)),
+        (Hardfork::Petersburg, ForkKind::Block(7280000)),
+        (Hardfork::Istanbul, ForkKind::Block(9069000)),
+        (Hardfork::Muirglacier, ForkKind::Block(9200000)),
+        (Hardfork::Berlin, ForkKind::Block(12244000)),
+        (Hardfork::London, ForkKind::Block(12965000)),
+        (Hardfork::ArrowGlacier, ForkKind::Block(13773000)),
+        (Hardfork::GrayGlacier, ForkKind::Block(15050000)),
+        (
+            Hardfork::MergeNetsplit,
+            ForkKind::TTD(TerminalTotalDifficulty::new(
+                U256::from(58750000000000000000000_u128),
+                Some(15537394),
+            )),
+        ),
+        (Hardfork::Latest, ForkKind::Block(15050000)),
     ]),
     dao_fork_support: true,
-    paris_block: Some(15537394),
-    paris_ttd: Some(U256::from(58750000000000000000000_u128)),
 });
```

Comment:
This is actually not necessary, and would lead to an incorrect forkid calculation. `Hardfork::MergeNetsplit` is a block-activated version of the merge for sepolia, and it is not used in mainnet:
 - https://github.com/ethereum/execution-specs/pull/564

Because of that, I don't think removing the `paris_ttd` here makes sense, although removing `paris_block` does.

Reply:
To clarify, are you saying that `Hardfork::MergeNetsplit` should be included for forkid calculation on Sepolia? With this PR, only hardforks declared with `ForkKind::Block` and `ForkKind::Time` are used to compute forkid.

I think it's nice to be able to get the Paris block with `ChainSpec::fork_block()`... Do you think it's ok to add `Hardfork::Paris` to solve all the naming issues?

Reply:
Yes, `Hardfork::MergeNetsplit` is actually already included for sepolia:
https://github.com/paradigmxyz/reth/blob/b354a17f6570cf534cb8d15fc4c9986bc24581ab/crates/primitives/src/chain_spec.rs#L57-L62

It should not be included as a key in the `hardforks` map for any other network, and should not really be used anywhere else.

We could add `Hardfork::Paris`, and add an entry with `ForkKind::TTD`, as long as we don't use it to calculate the forkid, which should only include block or time activated forks.

Reply:
We could even rename `Hardfork::MergeNetsplit` to `Hardfork::Paris`, and declare it as `(Hardfork::Paris, ForkKind::TDD)` for Mainnet and `(Hardfork::Paris, ForkKind::Block(1735371))` for Sepolia. wdyt?

Reply:
Let's change it to `Paris`, but make sure to add a note that for non-sepolia networks this should be used with a TTD, and _not_ with a block or timestamp.